packages/remix-dev/cli/migrate/migrations/replace-remix-imports/index.ts

```diff
@@ -105,12 +105,9 @@ export const replaceRemixImports: MigrationFunction = async ({
   pkg.update({
     dependencies: {
       ...depsToObject(otherDeps),
-      "@remix-run/react": remixVersionSpec,
-      [`@remix-run/${runtime}`]: remixVersionSpec,
-      ...(adapter ? { [`@remix-run/${adapter}`]: remixVersionSpec } : {}),
-      ...(remixServeInstalled
-        ? { [`@remix-run/serve`]: remixVersionSpec }
-        : {}),
+      ...depsToObject(
+        remixDeps.map(({ name }) => ({ name, versionSpec: remixVersionSpec }))
```

Comment:
We want to enforce that users who run this migration have exactly:
- dependencies: `@remix-run/react`, `@remix-run/{adapter}` || `@remix-run/serve`, `@remix-run/{runtime}`
- devDependencies: `@remix-run/dev`

So explicitly declaring each of these (and no other Remix deps) is a feature of the migration.